test/cqlpy/test_tablets.py

```diff
@@ -181,7 +181,35 @@ def test_describe_initial_tablets(cql, skip_without_tablets):
             "AND TABLETS = { 'initial' : 1 }"
     with new_test_keyspace(cql, ksdef) as keyspace:
         desc = cql.execute(f"DESCRIBE KEYSPACE {keyspace}")
-        assert "and tablets = {'initial': 1}" in desc.one().create_statement.lower()
+        assert "and tablets = {'enabled': true, 'initial': 1}" in desc.one().create_statement.lower()
+
+
+# Test to ensure metadata is printed when tablets are enabled during schema description, regardless of the use of WITH INTERNALS.
+#
+# Reproduces https://github.com/scylladb/scylladb/issues/22866
+@pytest.mark.parametrize("with_internals", [True, False])
+@pytest.mark.parametrize("tablets", [None, True, False])
+@pytest.mark.parametrize("initial", [None, 0, 7])
+def test_describe_tablets_internals(cql, with_internals, tablets, initial):
+    tablets_str = "'enabled': true" if tablets or tablets is None else "'enabled': false"
+    initial_str = f", 'initial': {initial}" if initial is not None else ""
+    if tablets is None and initial is None:
+        tablets_def = ""
+    else:
+        tablets_def = f" AND TABLETS = {{{tablets_str}{initial_str}}}"
+    ksdef = f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1' }}{tablets_def}"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        desc = cql.execute(f"DESCRIBE SCHEMA{" WITH INTERNALS" if with_internals else ""};")
```

Comment:
The usage of double quates for the inner string is really confusing.
How about calculating the optional `" WITH_INTERNALS"` string as a variable and use the latter in to formatted query string?

Reply:
Done